apps/reconciliation/lib/reconciliation.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Reconciliation do
+  @moduledoc """
+  This module will reconcile the trace file generated for each process
+  to a global trace 
+  """
+
+  import Emulation
+
+  def combine_trace(file_path) do
+    case File.ls(file_path) do
+      {:ok, files} ->
+        files
+        |> Enum.filter(fn x -> !String.starts_with?(x, ".") end)
+        |> Enum.flat_map(fn x -> parse_file(file_path, x) end)
+
+      {:error, reason} ->
+        IO.puts("Unable to locate the trace files with error #{reason}")
+    end
+  end
+
+  defp parse_file(file_path, file_name) do
+    case File.read("#{file_path}/#{file_name}") do
+      {:ok, f_content} ->
+        case Jason.decode(f_content) do
+          {:ok, file_content} ->
+            events = Map.fetch!(file_content, "events")
+
+          {:error, _} ->
+            IO.puts("Unexpected trace file format, unable to decode to Json")
+        end
+
+      {:error, _} ->
+        IO.puts("Unable to read the trace file content")
+    end
+  end
+
+  def get_file_path() do
+    System.get_env("TRACE_FILES")
+  end
+
+  defp make_vectors_equal_length(v1, v2) do
+    v1_add = for {k, _} <- v2, !Map.has_key?(v1, k), do: {k, 0}
+    Map.merge(v1, Enum.into(v1_add, %{}))
+  end
+
+  @before :before
+  @hafter :after
+  @concurrent :concurrent
+
+  defp compare_component(c1, c2) do
+    cond do
+      c1 < c2 -> @before
+      c1 > c2 -> @hafter
+      true -> @concurrent
+    end
+  end
+
+  def compare_vclock(event1, event2) do
+    v1 = get_timestamp(event1)
+    v2 = get_timestamp(event2)
+
+    v1 = make_vectors_equal_length(v1, v2)
+    v2 = make_vectors_equal_length(v2, v1)
+
+    compare_result = Map.values(Map.merge(v1, v2, fn _k, c1, c2 -> compare_component(c1, c2) end))
+
+    if Enum.any?(compare_result, fn x -> x == @before end) do
```

Comment:
I think it might be cleaner to merge this into a `cond do` pattern matching with:

```
cond do
Enum.any(...before) && Enum.any(....after) -> concurrent
Enum.any(...before) -> before
Enum.any(...after) -> after
```

What do you think?